### ISSUE TITLE
Document import assertions

### ISSUE THREAD

**Issue by Josh-Cena** [mdn/content#19220]
## What is the new suggestion?

Document [import assertions](https://github.com/tc39/proposal-import-assertions).

## Why is it important or useful?

[Already shipped in V8 for over a year.](https://v8.dev/features/import-assertions) JSON modules have been unflagged in Node so a lot of people are starting to rely on it. Need more information on its syntax and semantics.

Very importantly, import assertions do not change the semantics of the module itself—it cannot affect how the module is interpreted, nor can it be used as part of the cache key. Virtually all existing toolings, except JSON modules themselves (`assert { type: "json" }`), that try to extend import assertions are using them in a spec-incompliant way. (TBH, that to me means the proposal is designed poorly.) We need to call that out.

We just need a new section under https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import, but with so many additions to JavaScript modules in the future, it may make sense to start a new reference chapter like "modules", parallel to the existing "classes" and "functions" chapters...

## Other supporting information

- Spec draft: https://tc39.es/proposal-import-assertions/
- Proposal: https://github.com/tc39/proposal-import-assertions
- This is a sibling proposal of [JSON modules](https://github.com/tc39/proposal-json-modules), so we should probably mention that as well.
- Only after this proposal does `import()` have a second parameter and therefore optional trailing comma. `import("foo",)` works in Chrome but is a syntax error in Safari/Firefox. We need to add this to BCD and the trailing commas page: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas

**Comment by teoli2003** [comment]
Will this needs some mdn/browser-compat-data updates, or is it already listed there? (If so, under what key?)

**Comment by Josh-Cena** [comment]
I don't think there's BCD for this either, no.

**Comment by coderaiser** [comment]
Import assertions now called [import attributes](https://github.com/tc39/proposal-import-attributes) and they has other syntax:

```js
import json from "./foo.json" with { type: "json" };
import("foo.json", { with: { type: "json" } });
```

**Comment by Josh-Cena** [comment]
[I am indeed aware of that](https://github.com/mdn/content/pull/19588#issuecomment-1489191080). I will update the linked PR once the new syntax gets shipped in browsers and Chrome reaches a decision on what to do with the `assert` syntax.

**Comment by coderaiser** [comment]
@Josh-Cena update first commit and topic name in this case, is it outdated and no longer relevant..

> the new syntax gets shipped in browsers and Chrome reaches a decision on what to do with the assert syntax

I think they just remove it, since it was experimental and [stage-2](https://github.com/tc39/proposal-import-attributes) only, so it was a bad idea to implement it from the beginning. Anyways I think that both variants should exist in `MDN`, right now we have a strange situation I don't remember before (only with TypeScript and generators, but that was simple case I think):
- There is implemented experimental `stage-2` feature;
- Standard is changed (that is normal situation for `stage-2);

So what to do people who already used it? I think they should [process their code with a codemod](https://github.com/coderaiser/putout/tree/v29.11.1/packages/plugin-convert-assert-to-with#readme) and worry not.

But we will see what [v8 team will reach to](https://v8.dev/features/import-assertions).

**Comment by Josh-Cena** [comment]
Instead of letting this PR sit around forever, I'm going to close this and open a new issue when the new syntax and semantics get shipped in browsers. There will be significant changes from the currently documented behavior.
